src/Albany_Application.cpp

```diff
@@ -735,6 +735,71 @@ Application::finalSetUp(
   for (int i = 0; i < responses.size(); ++i) { responses[i]->postRegSetup(); }
 }
 
+template<typename Traits>
+void
+Application::setSideSetExtents(Teuchos::RCP<PHX::FieldManager<PHAL::AlbanyTraits>>& in_fm) const
+{
+  // get number of worksets
+  const auto& wsElNodeEqID = disc->getWsElNodeEqID();
+  unsigned int const numWorksets = wsElNodeEqID.size();
+
+  // compute largest workset size over all worksets for each sideset name
+  std::map<std::string, unsigned int> maxSideSetSizes;
+  for (unsigned int i = 0; i < numWorksets; ++i) { 
+    const LocalSideSetInfoList& sideSetView = disc->getSideSetViews(i);
+    for (auto it = sideSetView.begin(); it != sideSetView.end(); ++it) {
+      if (maxSideSetSizes.find(it->first) == maxSideSetSizes.end()) {
+        maxSideSetSizes[it->first] = 0;
+      }
+      unsigned int sideSetSize = it->second.size;
+      maxSideSetSizes[it->first] = std::max(maxSideSetSizes[it->first], sideSetSize);
+    }
+  }
+
+  // Iterate over tags and set extents for sideset fields
+  const auto& tags = in_fm->getFieldTagsForSizing<Traits>();
+  in_fm->buildDagForType<Traits>();
+  for (auto& t : tags) {
+    auto& t_dl = t->nonConstDataLayout();
+    std::vector<PHX::Device::size_type> t_dims;
+    t_dl.dimensions(t_dims);
+
+    // Check if dimension[0] is Side
+    std::string first_dim_name = t_dl.name(0);
+    std::string t_identifier = t_dl.identifier();
+    std::string sideSetName = t_identifier.substr(0, t_identifier.find("<"));
+
+    if (first_dim_name == "Side") {
+
+      t_dims[0] = maxSideSetSizes[sideSetName];
```

Comment:
It looks like adding this check has caused a few tests to fail. These tests are `SideSetLaplacian_3D`, `FO_ISMIP_TestC_Epetra`, and `FO_ISMIP_TestC_Tpetra`. 

It seems layouts are being constructed for side sets that don't have a corresponding `sideSetView`. Any ideas how this can happen or how to handle this case? If there is no `sideSetView`, there would have been an error in the first evaluator that operated on the side set so I assume nothing is being done with these side sets.

Reply:
I'm puzzled actually: why should the 1st chunk of `t_identifier` be a side set name? Shouldn't those be just field names?

Reply:
Field identifiers have the field names but these are identifiers for the underlying layouts. MDALayout allows for a prefix to be added to these identifiers so I attached the side set names so we could pair the layouts to their corresponding side set.

The format for MDALayout identifiers is as follows:

Prefix<Rank1, Rank2,...>(Extent1,Extent2,...)

Reply:
I'd have to look at the output of all side fields and side set views and the error output. It would seem to me that if there's a side field in the fm, there should be a max size for that side.

Reply:
> I'd have to look at the output of all side fields and side set views and the error output. It would seem to me that if there's a side field in the fm, there should be a max size for that side.

Here is an example output for one of the failing tests. I've also pushed my current commit so you can see the changes I made.

[log-ctest-dynamic-sizing.1012351.txt](https://github.com/SNLComputation/Albany/files/6918720/log-ctest-dynamic-sizing.1012351.txt)

Reply:
The log has `Adding sideset SideSet4 to maxSideSetSizes` but then throws with `SideSet4 was not found` so there's definitely something buggy in the logic. Can you run with one proc, print the full tag name, layout name and maybe the max size for SideSet4?

Reply:
I couldn't replicate with a single proc but I can with 2. I printed the requested information for each MPI rank and it looks like sideSet4 only gets added to maxSideSetSizes on rank 0 but rank 1 still sees a layout for sideSet4 in the field manager. I'm going to double check that the value for "numWorksets" is correct.

Here is the log:
[updated-log-dynamic-sizing.txt](https://github.com/SNLComputation/Albany/files/6919054/updated-log-dynamic-sizing.txt)

Reply:
can you print the tag name of the field that uses the layout too?

Reply:
Might be that the sideset is empty on a rank, and maybe the disc setup does not create side views for them?

Reply:
Here is the log with field tag names included:
[updated-log-dynamic-sizing.txt](https://github.com/SNLComputation/Albany/files/6919536/updated-log-dynamic-sizing.txt)

There were issues with the Coord Vec field previously which we fixed by checking the response field manager for field tags in addition to the other field managers. I'm still not sure what situation a field tag would show up in a field manager for a side set that never gets set up.

Reply:
It looks like in this particular case, SideSet4 is set as a BC which, I think, constructs the appropriate evaluator for all ranks (it looks like it only needs a mesh spec). But, maybe there's no side discretization on a rank because the sideset is empty (as Luca pointed out). In this case, the evaluator still runs but it's skipped:
```
if (workset.sideSetViews->find(sideSetName)==workset.sideSetViews->end()) return;
```
So, we need to apply the same check. For a given field side, loop over all sideSetViews. If the side doesn't exist, then skip setting the size. Then, you can keep the same error checks you currently have after this check.

Alternatively, we could eliminate the evaluators associated with this side in the fm but I think that might be a larger design choice.

Reply:
Actually, it seems like maxSideSetSizes provides the same functionality and the errors would be redundant. So maybe something like:
```
if (maxSideSetSizes.find(sideSetName) == maxSideSetSizes.end()) continue;
```
would be equivalent.